app/controllers/user_controller.rb

```diff
@@ -0,0 +1,51 @@
+ServiceDiscovery::App.controllers :users do
+  
+  post '/:user_id/address' do
+    get_create_address_params
+    address = create_address @input, params[:user_id]
+    status 201
+    {address_id: address.id}.to_json
+  end
+
+  get '/:user_id/address' do
+    addresses = Address.where('user_id' => params[:user_id])
+    status 201
+    {addresses: addresses}.to_json
+  end
+
+  put '/:user_id/update_address' do
+    get_update_address_params
+    user_id = get_valid_user_id params[:user_id]
+    update_address user_id
+    addresses = Address.where('user_id' => user_id)
+    status 201
+    {addresses: addresses}.to_json
+  end
+
+  post '/:user_id/delete_addresses' do
+    get_delete_addresses_params
+    Address.transaction do
+      Address.destroy_all('user_id' => params[:user_id], 'id' => @input[:address_ids])
+    end
+    addresses = Address.where('user_id' => params[:user_id])
+    status 201
```

Comment:
Here status is required as I am updating the record. I think return status is required but the status should be 200 as we are not creating any object